File: backend/internal/core/domain/entities/order.go
package entities

import "time"

type Order struct {
	ID            uint           `json:"id"`
	Title         string         `json:"title"`
	IsActive      bool           `json:"is_active"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	OrderMappings []OrderMapping `json:"order_mappings"`
}

// OrderMapping table (many-to-many between ListQueue and Order)
type OrderMapping struct {
	ID          uint      `json:"id"`
	ListQueueID uint      `json:"list_queue_id"`
	OrderID     uint      `json:"order_id"`
	Checked     bool      `json:"checked"`
	ListQueue   ListQueue `json:"-"`
	Order       Order     `json:"order"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/types/api/order.ts
export type OrderMapping = {
  id: number | string;
  order_id: number;
  checked: boolean;
  order: { id: number; title: string };
};

export type UpdateOrderInput = {
  order_id: number;
  list_queue_id: number;
  checked: boolean;
};

export type UpdateOrderNameInput = {
  order_id: number;
  title: string;
};
